game/scripts/vscripts/items/siege_mode.lua

```diff
@@ -58,6 +73,31 @@ end
 
 --------------------------------------------------------------------------------
 
+function modifier_item_siege_mode_siege:OnCreated( event )
+	local spell = self:GetAbility()
+
+	spell.mod = self
+
+	self.atkRange = spell:GetSpecialValueFor( "siege_attack_range" )
```

Comment:
If that's the case then there's no need to have all those `or`s... 

My vote is to add the refresh methods even if they're never called, it allows the modifier to be used in different ways if we rework the item without needing to be aware of that particular behavior.

explicit > implicit